database-commons/src/main/java/io/cdap/plugin/db/connector/DBSpecificPath.java

```diff
@@ -0,0 +1,134 @@
+ * A valid path can end with/without a slash, e.g. "/", "", "/database/schema/table/", "database/schema/table/" are all
+ * valid.
+ * A valid path should contain at most two parts for those DB that don't support schema (e.g. mysql) or three parts
+ * for those DB that support schema (e.g. postgresql, oracle, sqlserver) separated by slash. The first part is
+ * database name. The second part is schema name for those DB that support schema or table name for those DB that
+ * don't support schema. Third part is table name (only applicable for those DB that support schema). Any part is
+ * optional. e.g. "/" , "/database" , "/database/schema" (for those DB that support Schema), "database/schema/table"
+ * (for those DB that support Schema), "/database/table" (for those DB that don't support Schema)
+ * Consecutive slashes are not valid , it will be parsed as there is an empty string part between the slashes. e.g.
+ * "//a" will be parsed as database name as empty and schema name or table name as "a". Similarly "/a//" will be parsed
+ * as database name as "a" and table name or schema name as empty.
+ *
+ */
+public class DBSpecificPath implements DBConnectorPath {
+  private final boolean supportSchema;
+  private String database;
+  private String schema;
+  private String table;
+
+  public DBSpecificPath(String path, boolean supportSchema) {
+    this.supportSchema = supportSchema;
+    parsePath(path);
+  }
+
+  private void parsePath(String path) {
+    if (path == null) {
+      throw new IllegalArgumentException("Path should not be null.");
+    }
+
+    //remove heading "/" if exists
+    if (path.startsWith("/")) {
+      path = path.substring(1);
+    }
+
+    // both "" and "/" are taken as root path
+    if (path.isEmpty()) {
+      return;
+    }
+
+    if (path.endsWith("/")) {
+      path = path.substring(0, path.length() - 1);
+    }
+
+    String[] parts = path.split("/", -1);
+
+    // path should contain at most three parts : database, schema and table
+    // for those that don't support schema, path should contain at most two parts : database and table
+    if (parts.length > 3 || !supportSchema && parts.length == 3) {
+      throw new IllegalArgumentException(String.format("Path should not contain more than %d parts.",
+        supportSchema ? 3 : 2));
+    }
+
+    database = parts[0];
+    validateName("Database", database);
+    if (parts.length > 1) {
+      if (supportSchema) {
+        schema = parts[1];
+        validateName("Schema", schema);
+      } else {
+        table = parts[1];
+        validateName("Table", table);
+      }
+
+      if (parts.length == 3) {
+        table = parts[2];
+        validateName("Table", table);
+      }
+    }
+  }
+
+  private void validateName(String property, String name) {
+    if (name.isEmpty()) {
+      throw new IllegalArgumentException(
+        String.format("%s should not be empty.", property));
+    }
+  }
+
+  @Override
+  public boolean containDatabase() {
```

Comment:
Please add javadocs for public methods

Reply:
the is overriding the parent method which already have the javadoc. And the overriding doesn't change any semantics here.